### PR TITLE
OLS-510: Add CSV code markers to the CRD

### PR DESCRIPTION
**Pull request by vbelouso**
Signed-off-by: Vladimir Belousov <vbelouso@redhat.com>

## Description

Added CSV markers to improve the user experience in the OpenShift UI and data validation

## Type of change

- [ ] Refactor
- [x] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up dependent library

## Related Tickets & Documents

- Related Issue #
- Closes https://issues.redhat.com/browse/OLS-510

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [x] PR has passed all pre-merge test jobs.
- [x] If it is a core feature, I have added thorough tests.

## Testing
- Please provide detailed steps to perform tests related to this code change.
- How were the fix/results from this change verified? Please provide relevant screenshots or results.

### CODE REVIEW

File: api/v1alpha1/olsconfig_types.go [hunk 1]
```diff
@@ -43,51 +45,71 @@ type OLSConfigStatus struct {
...
 	DeploymentConfig DeploymentConfig `json:"deployment,omitempty"`
-	// Log level. Default: "INFO". Valid options are DEBUG, INFO, WARNING, ERROR and CRITICAL.
+	// Log level. Valid options are DEBUG, INFO, WARNING, ERROR and CRITICAL. Default: "INFO".
 	// +kubebuilder:validation:Enum=DEBUG;INFO;WARNING;ERROR;CRITICAL
 	// +kubebuilder:default=INFO
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Log level"
 	LogLevel string `json:"logLevel,omitempty"`
 	// Disable Authorization for OLS server. Default: "false"
 	// +kubebuilder:default=false
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Authorization",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:booleanSwitch"}
```
Comment (raptorsun) [inline api/v1alpha1/olsconfig_types.go:67]: `DisableAuthorization`

**Comment by bparees** [comment]
@vbelouso asked me about this PR on slack so i just wanted to weigh in to say i think the descriptors are a useful improvement to the UI that makes things look more polished, so i support adding them.

**Review by raptorsun — APPROVED** [review]
/lgtm

File: api/v1alpha1/olsconfig_types.go [hunk 2]
```diff
@@ -43,51 +45,71 @@ type OLSConfigStatus struct {
...
 	QueryFilters []QueryFiltersSpec `json:"queryFilters,omitempty"`
 }
 
 // DeploymentConfig defines the schema for overriding deployment of OLS instance.
 type DeploymentConfig struct {
 	// Defines the number of desired OLS pods. Default: "1"
 	// +kubebuilder:default=1
-	Replicas  *int32                       `json:"replicas,omitempty"`
+	// +kubebuilder:validation:Minimum=0
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Number of pods",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:podCount"}
```
[the remark this answers is not in the document]
Comment (raptorsun) [reply api/v1alpha1/olsconfig_types.go:113]:
```suggestion
	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Number of replica",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:podCount"}
```

**Pull request merged**